interface_classifier.py

```diff
@@ -96,78 +127,147 @@ def predict(self, temp=None, distance_cutoff=5.5, acc_threshold=0.05):
+        # Q: Why is this calling classify?
+        # base_path = os.path.dirname(os.path.realpath(__file__))
+        # prediction = os.popen(
+        #     os.path.join(base_path, "prodigy_cryst", "classify.py")
+        #     + " "
+        #     + " ".join(features)
+        # ).read()
+        model_f = Path(
+            Path(__file__).resolve().parent, "prodigy_cryst/data/classifier.sav"
+        )
+        # Calling this will raise some warning about modules that will be deprecated
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            with open(model_f, "rb") as fh:
+                model = pickle.load(fh)
+            proba = list(model.predict_proba([features])[0])
+            prediction = ["BIO", "XTAL"][proba.index(max(proba))], proba[0], proba[1]
         self.predicted_class = prediction
 
     def as_dict(self):
         return_dict = {
-            'structure': self.structure.id,
-            'selection': self.selection,
-            'ICs': len(self.ic_network),
-            'link_density': self.link_density,
-            'predicted_class': self.predicted_class
+            "structure": self.structure.id,
+            "selection": self.selection,
+            "ICs": len(self.ic_network),
+            "link_density": self.link_density,
+            "predicted_class": self.predicted_class,
         }
         return_dict.update(self.bins)
         return return_dict
 
-    def print_prediction(self, outfile='', quiet=False):
+    def print_prediction(self, outfile="", quiet=False):
         if outfile:
-            handle = open(outfile, 'w')
+            handle = open(outfile, "w")
         else:
             handle = sys.stdout
 
         if quiet:
-            handle.write('[+] {0}\t{1}\n'.format(self.structure.id, self.predicted_class))
+            handle.write(
+                "[+] {0}\t{1}\n".format(self.structure.id, self.predicted_class)
+            )
         else:
-            handle.write('[+] Selection: {0}\n'.format(', '.join(self.selection)))
-            handle.write('[+] No. of intermolecular contacts: {0}\n'.format(len(self.ic_network)))
-            handle.write('[+] No. of charged-charged contacts: {0}\n'.format(self.bins['CC']))
-            handle.write('[+] No. of charged-polar contacts: {0}\n'.format(self.bins['CP']))
-            handle.write('[+] No. of charged-apolar contacts: {0}\n'.format(self.bins['AC']))
-            handle.write('[+] No. of polar-polar contacts: {0}\n'.format(self.bins['PP']))
-            handle.write('[+] No. of apolar-polar contacts: {0}\n'.format(self.bins['AP']))
-            handle.write('[+] No. of apolar-apolar contacts: {0}\n'.format(self.bins['AA']))
-            handle.write('[+] Link density: {0:3.2f}\n'.format(self.link_density))
-            handle.write('[+] Class: {}\n'.format(self.predicted_class))
+            handle.write("[+] Selection: {0}\n".format(", ".join(self.selection)))
+            handle.write(
```

Comment:
there is one of those examples where enforcing 80 chars is actually strange. You can ignore the `flake8`  errors adding `# noqa: E501` at the end of the line. Don't know about `black`. :snake: :coffee:

Reply:
looks fine to me! 👀